### PR TITLE
Fix Deluge client critical bug (vs Deluge 1.3.15)

### PR DESCRIPTION
**Pull request by igor-makarov**
Hi, I fixed two issues with the Deluge client:
1. The base URL wasn't being constructed correctly (`json` and not `/json`).
2. The jobs list wasn't being notified to UI on main thread.

### CODE REVIEW

File: Sources/Torrent Interface/Front End/Deluge.m [hunk 1]
```diff
@@ -83,7 +83,7 @@ - (NSString *)getUserFriendlyAppendString
 
 - (NSString *)getURLAppendString
 {
-	return [[[FileHandler.sharedInstance webDataValueForKey:@"relative_path"] orSome:@""] stringByAppendingPathComponent:@"json"];
+	return [[[FileHandler.sharedInstance webDataValueForKey:@"relative_path"] orSome:@""] stringByAppendingPathComponent:@"/json"];
```
Comment (Qata) [inline Sources/Torrent Interface/Front End/Deluge.m]: That's weird, I thought `stringByAppendingPathComponent` would take care of there being a `/` there.

File: Sources/Connection Handler/TorrentJobChecker.m [hunk 2]
```diff
@@ -35,8 +35,10 @@ + (TorrentJobChecker *)sharedInstance
 
 - (void)updateTorrentClientWithJobsData
 {
-	[TorrentDelegate.sharedInstance.currentlySelectedClient handleTorrentJobs];
-	[NSNotificationCenter.defaultCenter postNotificationName:@"update_torrent_jobs_table" object:nil];
+    [TorrentDelegate.sharedInstance.currentlySelectedClient handleTorrentJobs];
+    dispatch_sync(dispatch_get_main_queue(), ^{
```
Comment (Qata) [inline Sources/Connection Handler/TorrentJobChecker.m]: Please change to `dispatch_async`.

**Comment by Qata** [comment]
Really great work! Please just address that one minor issue and I'll merge.

File: Sources/Torrent Interface/Front End/Deluge.m (hunk 1 above)
Comment (igor-makarov) [reply Sources/Torrent Interface/Front End/Deluge.m]: Well, the left operand is `@""`. Then it moves upstream to `TorrentClient.m` and there's no path appender there...

**Comment by igor-makarov** [comment]
I made that change you pointed out and also changed one more place where dispatch was blocking.

**Pull request merged**